scripts/clan_resources/freshkill.py

```diff
@@ -107,45 +58,29 @@ def remove_freshkill(self, amount, take_random: bool = False) -> None:
                 if it should be taken from the different sub-piles or not
         """
         if amount == 0:
-            return
+            return 0
         order = ["expires_in_1", "expires_in_2", "expires_in_3", "expires_in_4"]
         if take_random:
             random.shuffle(order)
         for key in order:
             amount = self.take_from_pile(key, amount)
+        return amount
 
-    def update_total_amount(self):
-        """
-        Update the total amount of the prey pile
-        """
-        self.total_amount = sum(self.pile.values())
+    @property
+    def total_amount(self):
+        return round(sum(self.pile.values()), 2)
 
     def _update_needed_food(self, living_cats: List[Cat]) -> None:
-        queen_dict, living_kits = get_alive_clan_queens(self.living_cats)
-        relevant_queens = []
-        # kits under 3 months are feed by the queen
-        for queen_id, their_kits in queen_dict.items():
-            queen = Cat.fetch_cat(queen_id)
-            if queen and (queen.outside or queen.status == "exiled"):
-                continue
-            young_kits = [kit for kit in their_kits if kit.moons < 3]
-            if len(young_kits) > 0:
-                relevant_queens.append(queen)
-        pregnant_cats = [
-            cat
-            for cat in living_cats
-            if "pregnant" in cat.injuries
-            and cat.ID not in queen_dict.keys()
-            and not cat.outside
-            and cat.status != "exiled"
-        ]
+
+        # handle kits and queens
+        clan_cats = self.handle_kits_and_queens(living_cats)
 
         # all normal status cats calculation
         needed_prey = sum(
             [
                 PREY_REQUIREMENT[cat.status]
                 for cat in living_cats
-                if cat.status not in ["newborn", "kitten", "exiled"] and not cat.outside
+                if cat.status not in ["kittypet", "rogue", "loner", "exiled", "former Clancat"] and not cat.outside
```

Comment:
I feel like `not cat.outside` should cover all of these statuses? Ideally, of course

Reply:
Unfortunately, "ideally" really is the operative word. This line is the number-one cause of test failures specifically because of that assumption. Because "outside" is a transient variable that technically depends on other variables to determine its state - but doesn't actually compute its state from them - it causes a lot of problems. The proper fix will be to tie `cat.outside` to its determining variables correctly (or just delete it as I'm really hoping to do). That said, that's outside the scope of this particular refactor.

Reply:
Though there _is_ a bug in this line, so I'll go back in and fix that.

Reply:
I see, I see. I was hoping that wasn't the case, but it's not surprising that it is, which is why I said "ideally" xD
Don't worry about that then.